src/shared/install.c

```diff
@@ -2790,24 +2790,34 @@ static int do_unit_file_disable(
 
         _cleanup_(install_context_done) InstallContext ctx = { .scope = scope };
         _cleanup_set_free_free_ Set *remove_symlinks_to = NULL;
+        InstallInfo *info;
+        bool has_install_info = false;
         int r;
 
         STRV_FOREACH(name, names) {
-                if (!unit_name_is_valid(*name, UNIT_NAME_ANY))
-                        return install_changes_add(changes, n_changes, -EUCLEAN, *name, NULL);
-
-                r = install_info_add(&ctx, *name, NULL, lp->root_dir, /* auxiliary= */ false, NULL);
+                r = install_info_discover(&ctx, lp, *name,
```

Comment:
This makes "disable" command accepts full path? Maybe, better to call `install_info_add()` and `install_info_traverse()`?